default.py

```diff
@@ -131,14 +308,39 @@ def _add_play_item(d):
         item.setInfo('video', info)
         is_folder = False
         url = get_url(action='play', session=sessionId, title=name.encode('utf-8'))
-        xbmcplugin.addDirectoryItem(_handle, url, item, is_folder)
+        xbmcplugin.addDirectoryItem(_kodihandle, url, item, is_folder)
     # If there are no sessions, the contentID is the sessionID
     for sessiond in content['Sessions']:
         _add_play_item(sessiond)
     if not content['Sessions']:
         _add_play_item(content)
 
-    xbmcplugin.endOfDirectory(_handle)
+    xbmcplugin.endOfDirectory(_kodihandle)
+
+# Play the url via the web url not api
+
+def play_web_session(url_from_search, content_id_from_search):
+    HEADERSFORPLAYER =  {
+        'User-Agent' :  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:52.0) Gecko/20100101 Firefox/52.0',
+        'Accept-Encoding' : 'gzip,deflate,br',
+        "x-requested-with":"XMLHttpRequest",
+        "Connection":"keep-alive",
+        "Accept":"*/*",
+        'Host' :'www.rightnowmedia.org',
+        'Referer' : url_from_search,
+    }
+    sessionInstance.readCookie()
+    url_response = sessionInstance.makeRequest(url_from_search + "#1", None, True, HEADERSFORPLAYER)
+    returnReg = re.search(".+session-contentId=\"([0-9]+)\"",url_response.text)
+    if returnReg is None:
+        returnReg = re.search(".+var contentId = ([0-9]+);.*",url_response.text)
+    if returnReg is None:
+        pass
+    else:
+        content_id = returnReg.group(1)
+    url = sessionInstance.makeRequest("https://www.rightnowmedia.org/Media/Player",{"contentId": str(content_id),'showPoster' : 'false' },True,HEADERSFORPLAYER)
```

Comment:
Yes. I  noticed that. I made a minor change to fix that so it'll error out if the content id is missing.